lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -105,22 +105,25 @@ LinearLayout makeCgaLayout(CTALayoutAttr layout) {
   return ret.transposeOuts(outDimNames);
 }
 
-// Shrinks the output set of a layout function while leaving the input set
-// unchanged, by making high-order inputs in inDimName map to the same output.
-// Attempts to shrink down to desiredSize, but this is not always possible just
-// by modifying one the specified input dimension.
+// For each out-dim d, ensure the layout's out-size (i.e. its codomain) is no
+// larger than shape[d].  Do this without changing the size of the layout's
+// inputs (i.e. leave its domain unchanged).
+//
+// This function is invariant to the order of the layout's input and output
+// dimensions.
 //
-// We do this by making the most-major inputs to the layout map to 0.  This
-// effectively duplicates data along that input dimension.  For example, this
-// layout has out-dim size 32:
+// We do this by making the largest value each output dimension 0 because
+// bases that map to a location larger than the input dimension is essentially
+// duplicating along that dimension.
+// For example, this layout has out-dim size 32, and suppose the shape is [8]:
 //
 //   L(register=1) = 8
 //   L(register=2) = 4
 //   L(register=4) = 1
 //   L(lane=1) = 2
 //   L(lane=2) = 16.
 //
-// If we shrink it to size 16 along the `lane` dimension, we set L(lane=2) to 0:
+// If we shrink it to size 16, we set L(lane=2) to 0:
```

Comment:
Above you say we're going to shrink it to have out-dim shape 8, but here we're shrinking it to 16?  I don't understand.